keras_hub/src/models/yolo_v8/ciou_loss.py

```diff
@@ -0,0 +1,86 @@
+import keras
+from keras import ops
+
+from keras_hub.src.bounding_box.iou import compute_ciou
+
+
+class CIoULoss(keras.losses.Loss):
+    """Implements the Complete IoU (CIoU) Loss
+
+    CIoU loss is an extension of GIoU loss, which further improves the IoU
+    optimization for object detection. CIoU loss not only penalizes the
+    bounding box coordinates but also considers the aspect ratio and center
+    distance of the boxes. The length of the last dimension should be 4 to
+    represent the bounding boxes.
+
+    Args:
+        bounding_box_format: a case-insensitive string (for example, "xyxy").
+            Each bounding box is defined by these 4 values. For detailed
+            information on the supported formats, see the [KerasCV bounding box
+            documentation](https://keras.io/api/keras_cv/bounding_box/formats/).
+        epsilon: A small value added to avoid division by zero and stabilize
+            calculations.
+
+    References:
+        - [CIoU paper](https://arxiv.org/pdf/2005.03572.pdf)
+
+    Example:
+    ```python
+    y_true = np.random.uniform(
+        size=(5, 10, 5),
+        low=0,
+        high=10)
+    y_pred = np.random.uniform(
+        (5, 10, 4),
+        low=0,
+        high=10)
+    loss = keras_hub.src.models.yolo_v8.ciou_loss.CIoULoss("xyxy")
+    loss(y_true, y_pred).numpy()
```

Comment:
the y_true and y_pred have different shapes - would this not result in a n error?